Framework/MDAlgorithms/test/CalculateCoverageDGSTest.h

```diff
@@ -98,25 +99,23 @@ class CalculateCoverageDGSTest : public CxxTest::TestSuite {
                (Mantid::PhysicalConstants::h * Mantid::PhysicalConstants::h);
     DblMatrix inverserubw(3, 3, true);
     inverserubw /= M_PI;
-    double phi = inputWorkspace->getInstrument()->getDetector(0)->getPhi();
-    double twoTheta =
-        inputWorkspace->getInstrument()->getDetector(0)->getTwoTheta(
-            sampPos, sourcePos * (-1.));
+    const auto &detectorInfo = inputWorkspace->detectorInfo();
+    const double phi = detectorInfo.detector(0).getPhi();
+    const double twoTheta = detectorInfo.twoTheta(0);
```

Comment:
`0` is the detector ID (`detid_t`), but `DetectorInfo` has an index as an argument (similar to the difference between spectrum number and workspace index). You need to translate the index via `DetectorInfo::indexOf()`.